src/main/java/uk/gov/hmcts/darts/common/service/impl/DataAnonymisationServiceImpl.java

```diff
@@ -97,8 +122,51 @@ public UserAccountEntity getUserAccount() {
         return userIdentity.getUserAccount();
     }
 
+    void tidyUpTransformedMediaEntities(UserAccountEntity userAccount, CourtCaseEntity courtCase) {
+        courtCase
+            .getHearings()
+            .stream()
+            .map(HearingEntity::getMediaRequests)
+            .flatMap(Collection::stream)
+            .distinct()
+            .peek(mediaRequestEntity -> expiredMediaRequest(userAccount, mediaRequestEntity))
+            .map(MediaRequestEntity::getTransformedMediaEntities)
```

Comment:
not sure adding getTransformedMediaEntities to MediaRequestEntity jsut for this is justified, might be better to just call repository method here.

Reply:
As the items are Lazy loaded it only gets loaded when we call getTransformedMediaEntities() so we should not have any performance impacts. 

This also helps link the objects conceptually in code simplifying logic and helping spring understand the relationship (useful when using things like QueryDSL)

Reply:
well it means you are having to wrap methods in transactional annotations to allow the lazy loading, which isn't great.
e.g. i can see you are adding another one here:-
https://github.com/hmcts/darts-api/pull/2044/files#diff-38e2fe2df168b40916a7e1fc72cc6313bbb56d64952a201c1ba12c27f223a7a0R42

Reply:
We only need to wrap the method if we want to make use of the Lazy loaded item :)

For that second transactional we want this one as we do not want to partially anonymise a case as this could lead to data not getting anonymised in the future or data sticking around in the blob stores